File: leetcode/solutions/0705.DesignHashSet/MyHashSet_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import MyHashSet from "./MyHashSet.ts";

Deno.test("0705. Design HashSet", () => {
  const hashSet = new MyHashSet();

  hashSet.add(1);
  hashSet.add(2);
  assertEquals(hashSet.contains(1), true);
  assertEquals(hashSet.contains(3), false);

  hashSet.add(2);
  assertEquals(hashSet.contains(2), true);

  hashSet.remove(2);
  assertEquals(hashSet.contains(2), false);
});

File: leetcode/solutions/0705.DesignHashSet/MyHashSet_test.go
package solution

import (
	"testing"
)

func Test_0705_DesignHashSet(t *testing.T) {
	t.Parallel()

	hashSet := Constructor()
	hashSet.Add(1)
	hashSet.Add(2)

	if !hashSet.Contains(1) {
		t.Errorf("got %#v want %#v", false, true)
	}
	if hashSet.Contains(3) {
		t.Errorf("got %#v want %#v", true, false)
	}

	hashSet.Add(2)

	if !hashSet.Contains(2) {
		t.Errorf("got %#v want %#v", false, true)
	}

	hashSet.Remove(2)

	if hashSet.Contains(2) {
		t.Errorf("got %#v want %#v", true, false)
	}
}
